website/blog/2025-04-20-vs-code-with-podman-desktop.md

```diff
@@ -0,0 +1,100 @@
+
+## VS Code: Your IDE for Container Development
+
+VS Code is a popular and versatile code editor that can be extended to enhance its functionality. For container development, several excellent extensions integrate seamlessly with Podman.
+
+## Prerequisites
+
+Before we begin, ensure you have the following installed:
+
+- **Podman:** Follow the installation instructions on the [official website](https://podman.io/).
+- **Podman Desktop:** Download and install Podman Desktop from the [official website](https://podman-desktop.io/downloads).
+- **Visual Studio Code:** Download and install VS Code from the [official website](https://code.visualstudio.com/download).
+
+## **VS Code Extensions**
+
+To integrate VS Code with Podman, we have 2 extensions as options:
+
+1. [**Microsoft’s Container Tools extension**](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-containers)**:** The "Container Tools" extension provides excellent support for container-related tasks, including building images, managing containers, and working with Containerfiles and Dockerfiles.
+2. [**Pod Manager**](https://marketplace.visualstudio.com/items/?itemName=dreamcatcher45.podmanager): This extension was created by one of the members of our community and is completely open source. It is designed to help you manage Podman containers, images, volumes, and networks directly from the VS Code interface.
+
+Let's install them one by one.
+
+### Option 1: Microsoft’s Container Tools extension
+
+To install the extension:
+
+1. Open VS Code.
+2. Click on the Extensions icon in the Activity Bar (or press `Ctrl+Shift+X` or `Cmd+Shift+X`).
+3. Search for "Container Tools" and install the extension by Microsoft.
+
+#### Configuring VS Code for Podman
+
+The Container Tools extension usually automatically detects Podman if Docker isn't running, by looking at the `DOCKER_HOST` environment variable. In Podman Desktop navigate to Settings \> Docker Compatibility \> Third-Party Tool Compatibility and make sure the option is enabled. It’s possible that if you installed through `brew` that this is not correctly set.
+
+![enabling docker compatibility in the settings](img/vs-code-podman/docker-compatibility.png)
+
+### Option 2: Pod Manager
+
+To install the extension:
+
+1. Open VS Code.
+2. Click on the Extensions icon in the Activity Bar (or press `Ctrl+Shift+X` or `Cmd+Shift+X`).
+3. Search for "Pod Manager" and install the extension by dreamcatcher45.
+
+## Using the VS Code Extensions
+
+Now that we have the extensions installed and configured, let's see how to use them.
+
+### Working with Containerfiles and Dockerfiles
+
+Both extensions provide syntax highlighting, code completion, and linting for Containerfiles and Dockerfiles. Open a Containerfile in VS Code, and you'll immediately benefit from these features.
+
+You can also build images directly from VS Code:
+
+1. Right-click on the Containerfile in the Explorer view.
+2. Select "Build Image".
```

Comment:
A big portion of the developers might also be just using the integrated terminal from VSCode.

Reply:
Yea that is true, but you don't need an extension in VS code to be able to do that. I really wanted to focus here how to use the extensions.

Reply:
It's not one or the other. In general developers are using the two. So how users who are using the CLI can also get a better experience is valuable.
Even if you focus on how to use the extensions, I'd still recommend trying to map to how most of the users are working. You could say something like "If you are working directly from the CLI, you'll see the built image appearing in the UI of the extension"...

Reply:
Makes sense, I added a line for that